src/config.py

```diff
@@ -155,14 +199,17 @@ def _add_to_pipeline(self, name: str, category: str, pipelines: List[str]):
             ):
                 self._config["service"]["pipelines"][pipeline][category].append(name)
 
-    def add_extension(self, name: str, extension_config: Dict[str, Any]) -> "Config":
-        """Add an extension to the config."""
-        if name not in self._config["service"]["extensions"]:
-            self._config["service"]["extensions"].append(name)
-        self._config["extensions"][name] = extension_config
-        return self
-
     @property
     def ports(self) -> List[int]:
         """Return the ports that are used in the Collector config."""
         return [port.value for port in Ports]
```

Comment:
We'd probably need something more sophisticated here: currently this would open all indicated ports at all times.